pkg/config/migration/multi_account.go

```diff
@@ -0,0 +1,124 @@
+package migration
+
+import (
+	"fmt"
+
+	"github.com/cli/go-gh/v2/pkg/config"
+)
+
+type CowardlyRefusalError struct {
+	Reason string
+}
+
+func (e CowardlyRefusalError) Error() string {
+	// Consider whether we should add a call to action here like "open an issue with the contents of your redacted hosts.yml"
+	return fmt.Sprintf("cowardly refusing to continue with multi account migration: %s", e.Reason)
+}
+
+var hostsKey = []string{"hosts"}
+
+// This migration exists to take a hosts section of the following structure:
+//
+//	github.com:
+//	  user: williammartin
+//	  git_protocol: https
+//	  editor: vim
+//	github.localhost:
+//	  user: monalisa
+//	  git_protocol: https
+//
+// We want this to migrate to something like:
+//
+// github.com:
+//   active_user: williammartin
+//	 users:
+//	   williammartin:
+//	     git_protocol: https
+//	     editor: vim
+//
+// github.localhost:
+//	 active_user: monalisa
+//     users:
+//	     monalisa:
+//		   git_protocol: https
+//
+// The reason for this is that we can then add new users under a host.
+
+type MultiAccount struct{}
```

Comment:
So, I put this down in the `config` package so it could live with this PR but I'm actually in two minds about where it belongs. The `config` exposed by `go-gh` is agnostic about its contents _except_ for the `hosts` key. In some ways it would be nice to abstract that knowledge and use some other generalised state to indicate when a file should be split but that's a topic for another time and too much work anyway.

Beyond that, the contents of the configuration are an _application_ level concern. That's when we provide meaning to the contents via the `AuthConfig`, or otherwise have expectations that we can read and write specific entries. The result of this is that perhaps this migration actually belongs in `cli/cli`. The seam is already there to support this.

Reply:
I love the idea of putting this in `cli/cli` rather than in `cli/go-gh`.